ddl/ddl_api.go

```diff
@@ -897,6 +905,33 @@ func (d *ddl) AlterTable(ctx context.Context, ident ast.Ident, specs []*ast.Alte
 	return nil
 }
 
+func (d *ddl) RebaseAutoID(ctx context.Context, ident ast.Ident, newBase int64) error {
+	is := d.GetInformationSchema()
+	schema, ok := is.SchemaByName(ident.Schema)
+	if !ok {
+		return infoschema.ErrDatabaseNotExists.GenByArgs(ident.Schema)
+	}
+	t, err := is.TableByName(ident.Schema, ident.Name)
+	if err != nil {
+		return errors.Trace(infoschema.ErrTableNotExists.GenByArgs(ident.Schema, ident.Name))
+	}
+	autoIncID, err := t.Allocator(ctx).NextGlobalAutoID(t.Meta().ID)
```

Comment:
We'd can put this logical to `onRebaseAutoID`。Because now `autoIncID` is n，its value may change when you rebase the global auto ID in `onRebaseAutoID`。

Reply:
There is another check in storage node to guarantee the value be set properly. The comparison here is trying to set a more reasonable value in table info.

Reply:
Seems that the global auto-id is not the current max id, for example, if the current max id is 1, the new base is 1000, then the next id inserted will be 5001. Is it desirable?

Reply:
It is our expected result. I'll add a test for it.

Reply:
Done. PTAL @lamxTyler